src/core_plugins/i18n/server/__tests__/i18n.js

```diff
@@ -0,0 +1,214 @@
+        const languageTag = 'en';
+        const expectedTranslationJson = {
+          'test_plugin_1-NO_SSL': 'Dont run the dev server using HTTPS',
+          'test_plugin_1-DEV': 'Run the server with development mode defaults',
+          'test_plugin_1-NO_RUN_SERVER': 'Dont run the dev server',
+          'test_plugin_1-HOME': 'Run along home now!',
+          'test_plugin_2-XXXXXX': 'This is XXXXXX string',
+          'test_plugin_2-YYYY_PPPP': 'This is YYYY_PPPP string',
+          'test_plugin_2-FFFFFFFFFFFF': 'This is FFFFFFFFFFFF string',
+          'test_plugin_2-ZZZ': 'This is ZZZ string'
+        };
+        return checkTranslations(expectedTranslationJson, languageTag);
+      });
+
+      it('should return the translations for de locale as registered' , function () {
+        const languageTag = 'de';
+        const expectedTranslationJson = {
+          'test_plugin_1-NO_SSL': 'Dont run the DE dev server using HTTPS',
+          'test_plugin_1-DEV': 'Run the DE server with development mode defaults'
+        };
+        return checkTranslations(expectedTranslationJson, languageTag);
+      });
+
+      it('should return the most recently registered translation for a key that has multiple translations' , function () {
+        i18n.registerTranslations(join(FIXTURES, 'translations', 'test_plugin_2', 'de.json'));
+        const languageTag = 'de';
+        const expectedTranslationJson = {
+          'test_plugin_1-NO_SSL': 'Dont run the DE dev server using HTTPS! I am regsitered afterwards!',
+          'test_plugin_1-DEV': 'Run the DE server with development mode defaults'
+        };
+        return checkTranslations(expectedTranslationJson, languageTag);
+      });
+
+    });
+
+    describe('getRegisteredTranslationLocales', function () {
+
+      it('should return all locales registered', function () {
+        const expectedLocales = ['en', 'de'];
+        return checkRegisteredLocales(expectedLocales);
+      });
+
+    });
+
+  });
+
+  describe('registerTranslations', function () {
+
+    it('should throw error when registering empty filename', function () {
+      return expect(i18n.registerTranslations).withArgs('').to.throwError();
+    });
+
+    it('should throw error when registering filename with no extension', function () {
+      return expect(i18n.registerTranslations).withArgs('file1').to.throwError();
+    });
+
+    it('should throw error when registering filename with non JSON extension', function () {
+      return expect(i18n.registerTranslations).withArgs('file1.txt').to.throwError();
+    });
+
+  });
+
+});
+
+function checkTranslationsForDefaultLocale(expectedTranslations) {
+  return i18n.getTranslationsForDefaultLocale()
+  .then(function (actualTranslations) {
+    expect(_.isEqual(actualTranslations, expectedTranslations)).to.be(true);
+  });
+}
+
+function checkTranslations(expectedTranslations, languageTags) {
+  return i18n.getTranslations(languageTags)
```

Comment:
Yeah, got caught between two stools. I have added the spread operator in calling the function and removed the array from the function definition. Works correctly now. Thanks for the feedback.